core/src/main/scala/cats/Foldable.scala

```diff
@@ -275,6 +275,9 @@ import Foldable.sentinel
    */
   def combineAll[A: Monoid](fa: F[A]): A = fold(fa)
 
+  def combineAllOption[A](fa: F[A])(implicit ev: Semigroup[A]): Option[A] =
+    if (isEmpty(fa)) None else ev.combineAllOption(iterable(fa))
```

Comment:
If we have iterable now, should we use it in fold for the default implementation? I think we should.

Reply:
Right, it could be implemented in terms of the Monoid's `combineAll` like
```scala
A.combineAll(iterable(fa))
```
this would allow to optimize the `fold` providing a specialization for `Monoid[A]`.
I'll add it in.